src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerEthSpec.scala

```diff
@@ -413,10 +413,10 @@ class JsonRpcControllerEthSpec
   }
 
   it should "eth_gasPrice" in new JsonRpcControllerFixture {
-    blockchainWriter
-      .storeBlock(Block(Fixtures.Blocks.Block3125369.header.copy(number = 42), Fixtures.Blocks.Block3125369.body))
-      .commit()
-    blockchain.saveBestKnownBlocks(ByteString.empty, 42)
+    private val block: Block =
+      Block(Fixtures.Blocks.Block3125369.header.copy(number = 42), Fixtures.Blocks.Block3125369.body)
+    blockchainWriter.storeBlock(block).commit()
+    blockchain.saveBestKnownBlocks(block.hash, 42)
```

Comment:
Since the block is now available, replace 42 by block.number.

Actually, I wonder if we shouldn't provide a blockchain.saveBestKnownBlockInfo(block: Block) method instead of / side by side with the saveBestKnowBlocks(hash, number) (and eventually rename it to saveBestKnowBlockInfo too, because as it is, I don't see it related to multiple block*s*, but only to on block*Info*)
I raise this as most of the time, I see the call is done by extracting values from a given block, and at the blockchain level, I don't see any evil with directly handling a block (no added dependency)

Reply:
There is already a `save` function that does something similar. I'm not sure why some test use it or not so I kept the same structure. But I think once we have a proper consensus and a central point where the best chain is decided it will be clearer what to do about that.